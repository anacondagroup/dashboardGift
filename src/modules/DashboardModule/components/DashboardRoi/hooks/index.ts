export * from './useRoiDashboardProtector';
export * from './useRoiTable/useRoiTable';
