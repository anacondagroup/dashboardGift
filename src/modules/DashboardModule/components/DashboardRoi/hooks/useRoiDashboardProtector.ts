import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';

import {
  getIsSfConnectionStateLoaded,
  getSfConnectionState,
} from '../../../../SettingsModule/store/organisation/integrations/workato/integrations/integrations.selectors';
import { ROI_ROUTES } from '../routePaths';
import { fetchSfConnectionState } from '../../../../SettingsModule/store/organisation/integrations/workato/integrations/integrations.actions';

export const useRoiDashboardProtector = (): {
  isLoaded: boolean;
  isSalesforceConnected: boolean;
  dashboard: string;
} => {
  const dispatch = useDispatch();

  const sfConnectionState = useSelector(getSfConnectionState);
  const isLoaded = useSelector(getIsSfConnectionStateLoaded);

  const dashboard = sfConnectionState === 'connected' ? ROI_ROUTES.FUNNEL : ROI_ROUTES.REPORTING;

  useEffect(() => {
    dispatch(fetchSfConnectionState());
  }, [dispatch]);

  return {
    isLoaded,
    isSalesforceConnected: sfConnectionState === 'connected',
    dashboard,
  };
};
