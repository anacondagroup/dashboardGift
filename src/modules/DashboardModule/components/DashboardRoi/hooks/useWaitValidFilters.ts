import { useEffect, useRef } from 'react';
import { TGlobalRoiFilters } from '@alycecom/services';

export const useWaitValidFilters = (filters: TGlobalRoiFilters): boolean => {
  const wasInitialRequestMadeRef = useRef(false);

  useEffect(() => {
    if (filters.teamIds.length) {
      wasInitialRequestMadeRef.current = true;
    }
  }, [filters.teamIds]);

  return wasInitialRequestMadeRef.current ? false : filters.teamIds.length === 0;
};
