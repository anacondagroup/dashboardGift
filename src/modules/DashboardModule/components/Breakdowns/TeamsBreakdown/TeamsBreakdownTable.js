import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { DISPLAY_DATE_FORMAT, TableCellTooltip, TableLoadingLabel } from '@alycecom/ui';
import { TrackEvent } from '@alycecom/services';
import { Link as RouterLink } from 'react-router-dom';
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  TableSortLabel,
  TableFooter,
  TablePagination,
  Typography,
} from '@mui/material';
import moment from 'moment';

import { teamBreakdownShape } from '../../../shapes/teamBreakdown.shape';
import { usePagination } from '../../../../../hooks/usePagination';

export const TeamsBreakdownTableComponent = ({
  renderToolbar,
  items,
  onSort,
  dir,
  sort,
  isLoading,
  linkQueryParams,
  page,
  onPageChange,
}) => {
  const { trackEvent } = TrackEvent.useTrackEvent();
  const rowsPerPage = 10;
  const [teams, showPagination, emptyRows, showEmptyRows] = usePagination(page, rowsPerPage, items, isLoading);

  const pageChangeHandler = useCallback((_, nextPage) => onPageChange(nextPage), [onPageChange]);

  return (
    <>
      {renderToolbar()}
      <Table padding="none">
        <TableHead>
          <TableRow>
            <TableCell>
              <TableSortLabel direction={dir} active={sort === 'name'} onClick={() => onSort('name')}>
                team name
              </TableSortLabel>
            </TableCell>
            <TableCell align="right">
              <TableSortLabel direction={dir} active={sort === 'gifts_sent'} onClick={() => onSort('gifts_sent')}>
                gifts sent
              </TableSortLabel>
            </TableCell>
            <TableCell align="right">
              <TableSortLabel direction={dir} active={sort === 'gifts_viewed'} onClick={() => onSort('gifts_viewed')}>
                gifts viewed
              </TableSortLabel>
            </TableCell>
            <TableCell align="right">
              <TableSortLabel
                direction={dir}
                active={sort === 'gifts_accepted'}
                onClick={() => onSort('gifts_accepted')}
              >
                gifts accepted
              </TableSortLabel>
            </TableCell>
            <TableCell align="right">
              <TableSortLabel
                direction={dir}
                active={sort === 'meetings_booked'}
                onClick={() => onSort('meetings_booked')}
              >
                meeting booked
              </TableSortLabel>
            </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {teams.map(team => (
            <TableRow key={team.id}>
              <TableCell component="th" scope="row">
                <TableLoadingLabel
                  maxWidth={400}
                  pr={2}
                  isLoading={isLoading}
                  render={() => (
                    <TableCellTooltip
                      renderLabel={() => (
                        <RouterLink
                          onClick={() =>
                            trackEvent('Enterprise dashboard â€” link from teams page to single team view', {
                              team_id: team.id,
                            })
                          }
                          to={`/teams/${team.id}?${linkQueryParams}`}
                        >
                          {team.name}
                          {Boolean(team.archivedAt) && (
                            <Typography display="inline" noWrap color="#99A1BD">
                              &nbsp;(Archived {moment(team.archivedAt).format(DISPLAY_DATE_FORMAT)})
                            </Typography>
                          )}
                        </RouterLink>
                      )}
                      title={team.name}
                    />
                  )}
                />
              </TableCell>
              <TableCell align="right">
                <TableLoadingLabel
                  align="right"
                  pl={2}
                  maxWidth={190}
                  isLoading={isLoading}
                  render={() => team.giftsSent}
                />
              </TableCell>
              <TableCell align="right">
                <TableLoadingLabel
                  align="right"
                  pl={2}
                  maxWidth={190}
                  isLoading={isLoading}
                  render={() => team.giftsViewed}
                />
              </TableCell>
              <TableCell align="right">
                <TableLoadingLabel
                  align="right"
                  pl={2}
                  maxWidth={190}
                  isLoading={isLoading}
                  render={() => team.giftsAccepted}
                />
              </TableCell>
              <TableCell align="right">
                <TableLoadingLabel
                  align="right"
                  pl={2}
                  maxWidth={190}
                  isLoading={isLoading}
                  render={() => team.meetingsBooked}
                />
              </TableCell>
            </TableRow>
          ))}
          {showEmptyRows > 0 && (
            <TableRow style={{ height: 48 * emptyRows }}>
              <TableCell colSpan={12} />
            </TableRow>
          )}
        </TableBody>
        {showPagination && (
          <TableFooter>
            <TableRow>
              <TablePagination
                rowsPerPageOptions={[rowsPerPage]}
                colSpan={12}
                count={items.length}
                rowsPerPage={rowsPerPage}
                page={page}
                SelectProps={{
                  native: true,
                }}
                onPageChange={pageChangeHandler}
              />
            </TableRow>
          </TableFooter>
        )}
      </Table>
    </>
  );
};

TeamsBreakdownTableComponent.propTypes = {
  sort: PropTypes.string.isRequired,
  dir: PropTypes.string.isRequired,
  renderToolbar: PropTypes.func.isRequired,
  onSort: PropTypes.func.isRequired,
  items: PropTypes.arrayOf(teamBreakdownShape).isRequired,
  onPageChange: PropTypes.func.isRequired,
  page: PropTypes.number.isRequired,
  isLoading: PropTypes.bool,
  linkQueryParams: PropTypes.string,
};

TeamsBreakdownTableComponent.defaultProps = {
  linkQueryParams: '',
  isLoading: true,
};

export default TeamsBreakdownTableComponent;
