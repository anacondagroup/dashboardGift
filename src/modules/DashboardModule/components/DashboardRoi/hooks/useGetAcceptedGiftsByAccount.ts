import { TAcceptedGiftByEmailDomain, useGetAcceptedGiftsByAccountQuery } from '@alycecom/services';

import { transformRoiFilters } from '../utils/transformRoiFilters';

import { TDebouncedFilters } from './useRoiTable/useRoiTable';
import { useWaitValidFilters } from './useWaitValidFilters';

export type TUseGetAcceptedGiftsByAccountParams = {
  accountId: string;
  filters: TDebouncedFilters<TAcceptedGiftByEmailDomain>;
};

export const useGetAcceptedGiftsByAccount = ({
  accountId,
  filters,
}: TUseGetAcceptedGiftsByAccountParams): ReturnType<typeof useGetAcceptedGiftsByAccountQuery> & {
  isWaitingForFilters: boolean;
} => {
  const isWaitingForFilters = useWaitValidFilters(filters);
  const queryResult = useGetAcceptedGiftsByAccountQuery(
    {
      filters: transformRoiFilters(filters),
      accountId,
    },
    { skip: isWaitingForFilters },
  );

  return {
    ...queryResult,
    isWaitingForFilters,
  };
};
