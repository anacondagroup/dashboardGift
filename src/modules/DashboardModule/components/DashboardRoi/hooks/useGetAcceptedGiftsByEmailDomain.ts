import { TAcceptedGiftByEmailDomain, useGetAcceptedGiftsByEmailDomainQuery } from '@alycecom/services';

import { transformRoiFilters } from '../utils/transformRoiFilters';

import { TDebouncedFilters } from './useRoiTable/useRoiTable';
import { useWaitValidFilters } from './useWaitValidFilters';

export type TUseGetAcceptedGiftsByEmailDomainParams = {
  emailDomain: string;
  filters: TDebouncedFilters<TAcceptedGiftByEmailDomain>;
};

export const useGetAcceptedGiftsByEmailDomain = ({
  emailDomain,
  filters,
}: TUseGetAcceptedGiftsByEmailDomainParams): ReturnType<typeof useGetAcceptedGiftsByEmailDomainQuery> & {
  isWaitingForFilters: boolean;
} => {
  const isWaitingForFilters = useWaitValidFilters(filters);
  const queryResult = useGetAcceptedGiftsByEmailDomainQuery(
    {
      filters: transformRoiFilters(filters),
      emailDomain,
    },
    { skip: isWaitingForFilters },
  );

  return {
    ...queryResult,
    isWaitingForFilters,
  };
};
