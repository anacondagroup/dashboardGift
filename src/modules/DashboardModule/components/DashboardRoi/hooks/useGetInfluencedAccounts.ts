import { TInfluencedAccounts, useGetInfluencedAccountsQuery } from '@alycecom/services';

import { transformRoiFilters } from '../utils/transformRoiFilters';

import { TDebouncedFilters } from './useRoiTable/useRoiTable';
import { useWaitValidFilters } from './useWaitValidFilters';

export const useGetInfluencedAccounts = (
  filters: TDebouncedFilters<TInfluencedAccounts>,
): ReturnType<typeof useGetInfluencedAccountsQuery> & {
  isWaitingForFilters: boolean;
} => {
  const isWaitingForFilters = useWaitValidFilters(filters);
  const queryResult = useGetInfluencedAccountsQuery(transformRoiFilters(filters), { skip: isWaitingForFilters });

  return {
    ...queryResult,
    isWaitingForFilters,
  };
};
