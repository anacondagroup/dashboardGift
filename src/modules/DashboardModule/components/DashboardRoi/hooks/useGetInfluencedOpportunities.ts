import { TInfluencedOpportunities, useGetInfluencedOpportunitiesQuery } from '@alycecom/services';

import { transformRoiFilters } from '../utils/transformRoiFilters';

import { TDebouncedFilters } from './useRoiTable/useRoiTable';
import { useWaitValidFilters } from './useWaitValidFilters';

export const useGetInfluencedOpportunities = (
  filters: TDebouncedFilters<TInfluencedOpportunities>,
): ReturnType<typeof useGetInfluencedOpportunitiesQuery> & {
  isWaitingForFilters: boolean;
} => {
  const isWaitingForFilters = useWaitValidFilters(filters);
  const queryResult = useGetInfluencedOpportunitiesQuery(transformRoiFilters(filters), { skip: isWaitingForFilters });

  return {
    ...queryResult,
    isWaitingForFilters,
  };
};
