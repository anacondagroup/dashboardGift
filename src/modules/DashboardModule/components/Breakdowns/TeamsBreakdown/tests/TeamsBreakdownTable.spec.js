import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import { TrackEvent } from '@alycecom/services';

import { TeamsBreakdownTableComponent } from '../TeamsBreakdownTable';
import { usePagination } from '../../../../../../hooks/usePagination';

jest.mock('@mui/material/Table', () => 'Table');
jest.mock('@mui/material/TableBody', () => 'TableBody');
jest.mock('@mui/material/TableCell', () => 'TableCell');
jest.mock('@mui/material/TableHead', () => 'TableHead');
jest.mock('@mui/material/TableRow', () => 'TableRow');
jest.mock('@mui/material/TableSortLabel', () => 'TableSortLabel');
jest.mock('@mui/material/TableFooter', () => 'TableFooter');
jest.mock('@mui/material/TablePagination', () => 'TablePagination');
jest.mock('react-router-dom', () => ({ Link: 'RouterLick' }));
jest.mock('../../../../../../hooks/usePagination');
jest.mock('@alycecom/modules', () => ({
  TableCellTooltip: 'TableCellTooltip',
  TrackEvent: {
    useTrackEvent: jest.fn(() => ({
      trackEvent: jest.fn(),
    })),
  },
}));
jest.mock('@alycecom/services', () => ({
  TableCellTooltip: 'TableCellTooltip',
  TrackEvent: {
    useTrackEvent: jest.fn(),
  },
}));
jest.mock('@alycecom/hooks', () => ({
  useRouting: jest.fn(),
  useReferrerUrl: jest.fn(),
}));

describe('TeamsBreakdownTable', () => {
  let props;
  beforeEach(() => {
    props = {
      isLoading: true,
      renderToolbar: () => <div data-toolbar="" />,
      onSort: () => {},
      dir: 'asc',
      sort: '',
      linkQueryParams: 'param=123',
      items: [],
      page: 0,
      onPageChange: jest.fn(),
    };
    TrackEvent.useTrackEvent.mockReset().mockImplementation(() => ({
      trackEvent: jest.fn(),
    }));
    usePagination.mockReset();
  });

  test('render while items are loading', () => {
    usePagination.mockReturnValue([[], false, 10, true]);

    const tree = create(shallow(<TeamsBreakdownTableComponent {...props} />));
    expect(usePagination).toBeCalledWith(0, 10, [], true);
    expect(tree).toMatchSnapshot();
  });

  test('render with teams', () => {
    const teams = Array.from({ length: 10 }, (_, id) => ({
      giftsSent: `giftsSent${id}`,
      giftsViewed: `giftsViewed${id}`,
      giftsAccepted: `giftsAccepted${id}`,
      meetingsBooked: `meetingsBooked${id}`,
      id,
    }));
    props.isLoading = false;
    usePagination.mockReturnValue([teams, true, 10, false]);

    const tree = create(shallow(<TeamsBreakdownTableComponent {...props} />));
    expect(usePagination).toBeCalledWith(0, 10, [], false);
    expect(tree).toMatchSnapshot();
  });
});
