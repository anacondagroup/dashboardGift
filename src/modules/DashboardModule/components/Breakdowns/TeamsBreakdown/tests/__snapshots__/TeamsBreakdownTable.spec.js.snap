// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamsBreakdownTable render while items are loading 1`] = `
Array [
  <div
    data-toolbar=""
  />,
  <Table
    padding="none"
  >
    <TableHead>
      <TableRow>
        <TableCell>
          <TableSortLabel
            active={false}
            direction="asc"
            onClick={[Function]}
          >
            team name
          </TableSortLabel>
        </TableCell>
        <TableCell
          align="right"
        >
          <TableSortLabel
            active={false}
            direction="asc"
            onClick={[Function]}
          >
            gifts sent
          </TableSortLabel>
        </TableCell>
        <TableCell
          align="right"
        >
          <TableSortLabel
            active={false}
            direction="asc"
            onClick={[Function]}
          >
            gifts viewed
          </TableSortLabel>
        </TableCell>
        <TableCell
          align="right"
        >
          <TableSortLabel
            active={false}
            direction="asc"
            onClick={[Function]}
          >
            gifts accepted
          </TableSortLabel>
        </TableCell>
        <TableCell
          align="right"
        >
          <TableSortLabel
            active={false}
            direction="asc"
            onClick={[Function]}
          >
            meeting booked
          </TableSortLabel>
        </TableCell>
      </TableRow>
    </TableHead>
    <TableBody>
      <TableRow
        style={
          Object {
            "height": 480,
          }
        }
      >
        <TableCell
          colSpan={12}
        />
      </TableRow>
    </TableBody>
  </Table>,
]
`;

exports[`TeamsBreakdownTable render with teams 1`] = `
Array [
  <div
    data-toolbar=""
  />,
  <Table
    padding="none"
  >
    <TableHead>
      <TableRow>
        <TableCell>
          <TableSortLabel
            active={false}
            direction="asc"
            onClick={[Function]}
          >
            team name
          </TableSortLabel>
        </TableCell>
        <TableCell
          align="right"
        >
          <TableSortLabel
            active={false}
            direction="asc"
            onClick={[Function]}
          >
            gifts sent
          </TableSortLabel>
        </TableCell>
        <TableCell
          align="right"
        >
          <TableSortLabel
            active={false}
            direction="asc"
            onClick={[Function]}
          >
            gifts viewed
          </TableSortLabel>
        </TableCell>
        <TableCell
          align="right"
        >
          <TableSortLabel
            active={false}
            direction="asc"
            onClick={[Function]}
          >
            gifts accepted
          </TableSortLabel>
        </TableCell>
        <TableCell
          align="right"
        >
          <TableSortLabel
            active={false}
            direction="asc"
            onClick={[Function]}
          >
            meeting booked
          </TableSortLabel>
        </TableCell>
      </TableRow>
    </TableHead>
    <TableBody>
      <TableRow>
        <TableCell
          component="th"
          scope="row"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-4 MuiBox-root css-1wizxrd"
          >
            <RouterLick
              onClick={[Function]}
              to="/teams/0?param=123"
            />
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-6 MuiBox-root css-mqotvu"
          >
            giftsSent0
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-8 MuiBox-root css-mqotvu"
          >
            giftsViewed0
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-10 MuiBox-root css-mqotvu"
          >
            giftsAccepted0
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-12 MuiBox-root css-mqotvu"
          >
            meetingsBooked0
          </div>
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          component="th"
          scope="row"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-14 MuiBox-root css-1wizxrd"
          >
            <RouterLick
              onClick={[Function]}
              to="/teams/1?param=123"
            />
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-16 MuiBox-root css-mqotvu"
          >
            giftsSent1
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-18 MuiBox-root css-mqotvu"
          >
            giftsViewed1
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-20 MuiBox-root css-mqotvu"
          >
            giftsAccepted1
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-22 MuiBox-root css-mqotvu"
          >
            meetingsBooked1
          </div>
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          component="th"
          scope="row"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-24 MuiBox-root css-1wizxrd"
          >
            <RouterLick
              onClick={[Function]}
              to="/teams/2?param=123"
            />
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-26 MuiBox-root css-mqotvu"
          >
            giftsSent2
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-28 MuiBox-root css-mqotvu"
          >
            giftsViewed2
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-30 MuiBox-root css-mqotvu"
          >
            giftsAccepted2
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-32 MuiBox-root css-mqotvu"
          >
            meetingsBooked2
          </div>
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          component="th"
          scope="row"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-34 MuiBox-root css-1wizxrd"
          >
            <RouterLick
              onClick={[Function]}
              to="/teams/3?param=123"
            />
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-36 MuiBox-root css-mqotvu"
          >
            giftsSent3
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-38 MuiBox-root css-mqotvu"
          >
            giftsViewed3
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-40 MuiBox-root css-mqotvu"
          >
            giftsAccepted3
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-42 MuiBox-root css-mqotvu"
          >
            meetingsBooked3
          </div>
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          component="th"
          scope="row"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-44 MuiBox-root css-1wizxrd"
          >
            <RouterLick
              onClick={[Function]}
              to="/teams/4?param=123"
            />
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-46 MuiBox-root css-mqotvu"
          >
            giftsSent4
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-48 MuiBox-root css-mqotvu"
          >
            giftsViewed4
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-50 MuiBox-root css-mqotvu"
          >
            giftsAccepted4
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-52 MuiBox-root css-mqotvu"
          >
            meetingsBooked4
          </div>
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          component="th"
          scope="row"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-54 MuiBox-root css-1wizxrd"
          >
            <RouterLick
              onClick={[Function]}
              to="/teams/5?param=123"
            />
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-56 MuiBox-root css-mqotvu"
          >
            giftsSent5
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-58 MuiBox-root css-mqotvu"
          >
            giftsViewed5
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-60 MuiBox-root css-mqotvu"
          >
            giftsAccepted5
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-62 MuiBox-root css-mqotvu"
          >
            meetingsBooked5
          </div>
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          component="th"
          scope="row"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-64 MuiBox-root css-1wizxrd"
          >
            <RouterLick
              onClick={[Function]}
              to="/teams/6?param=123"
            />
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-66 MuiBox-root css-mqotvu"
          >
            giftsSent6
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-68 MuiBox-root css-mqotvu"
          >
            giftsViewed6
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-70 MuiBox-root css-mqotvu"
          >
            giftsAccepted6
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-72 MuiBox-root css-mqotvu"
          >
            meetingsBooked6
          </div>
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          component="th"
          scope="row"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-74 MuiBox-root css-1wizxrd"
          >
            <RouterLick
              onClick={[Function]}
              to="/teams/7?param=123"
            />
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-76 MuiBox-root css-mqotvu"
          >
            giftsSent7
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-78 MuiBox-root css-mqotvu"
          >
            giftsViewed7
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-80 MuiBox-root css-mqotvu"
          >
            giftsAccepted7
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-82 MuiBox-root css-mqotvu"
          >
            meetingsBooked7
          </div>
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          component="th"
          scope="row"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-84 MuiBox-root css-1wizxrd"
          >
            <RouterLick
              onClick={[Function]}
              to="/teams/8?param=123"
            />
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-86 MuiBox-root css-mqotvu"
          >
            giftsSent8
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-88 MuiBox-root css-mqotvu"
          >
            giftsViewed8
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-90 MuiBox-root css-mqotvu"
          >
            giftsAccepted8
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-92 MuiBox-root css-mqotvu"
          >
            meetingsBooked8
          </div>
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          component="th"
          scope="row"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-94 MuiBox-root css-1wizxrd"
          >
            <RouterLick
              onClick={[Function]}
              to="/teams/9?param=123"
            />
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-96 MuiBox-root css-mqotvu"
          >
            giftsSent9
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-98 MuiBox-root css-mqotvu"
          >
            giftsViewed9
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-100 MuiBox-root css-mqotvu"
          >
            giftsAccepted9
          </div>
        </TableCell>
        <TableCell
          align="right"
        >
          <div
            className="makeStyles-cell-2 makeStyles-cell-102 MuiBox-root css-mqotvu"
          >
            meetingsBooked9
          </div>
        </TableCell>
      </TableRow>
    </TableBody>
    <TableFooter>
      <TableRow>
        <TablePagination
          SelectProps={
            Object {
              "native": true,
            }
          }
          colSpan={12}
          count={0}
          onPageChange={[Function]}
          page={0}
          rowsPerPage={10}
          rowsPerPageOptions={
            Array [
              10,
            ]
          }
        />
      </TableRow>
    </TableFooter>
  </Table>,
]
`;
